kubernetes/samples/scripts/domain-lifecycle/rollCluster.sh

```diff
@@ -0,0 +1,120 @@
+# !/bin/sh
+# Copyright (c) 2021, Oracle and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+#
+
+script="${BASH_SOURCE[0]}"
+scriptDir="$( cd "$( dirname "${script}" )" && pwd )"
+source ${scriptDir}/helper.sh
+if [ "${debug}" == "true" ]; then set -x; fi;
+
+function usage() {
+
+  cat << EOF
+
+  This script performs rolling restart of a WebLogic cluster server pods in a domain by updating the
+  value of 'spec.clusters[<cluster-name>].restartVersion' attribute of the domain resource. This change
+  will cause the operator to perform a rolling restart of the WebLogic cluster server pods.
+ 
+  Usage:
+ 
+    $(basename $0) -c mycluster [-n mynamespace] [-d mydomainuid] [-r restartVersion] [-m kubecli]
+  
+    -c <cluster-name>   : Cluster name parameter is required.
+
+    -d <domain_uid>     : Domain unique-id. Default is 'sample-domain1'.
+
+    -n <namespace>      : Domain namespace. Default is 'sample-domain1-ns'.
+
+    -r <restartVersion> : Restart version. If this parameter is not provided, then the script will generate the
+                          'restartVersion' value of the cluster by incrementing the existing value. If the
+                          'restartVersion' value doesn't exist for the cluster then it will increment the domain
+                          'restartVersion'. If the domain 'restartVersion' also doesn't exist or the effective
+                          value is non-numeric, then the script will set the 'restartVersion' value to '1'.
+
+    -m <kubernetes_cli> : Kubernetes command line interface. Default is 'kubectl' if KUBERNETES_CLI env
+                          variable is not set. Otherwise default is the value of KUBERNETES_CLI env variable.
+
+    -v <verbose_mode>   : Enables verbose mode. Default is 'false'.
+
+    -h                  : This help.
+   
+EOF
+exit $1
+}
+
+kubernetesCli=${KUBERNETES_CLI:-kubectl}
+clusterName=""
+domainUid="sample-domain1"
+domainNamespace="sample-domain1-ns"
+verboseMode=false
+patchJson=""
+restartVersion=""
+
+while getopts "vc:n:m:d:r:h" opt; do
+  case $opt in
+    c) clusterName="${OPTARG}"
+    ;;
+    n) domainNamespace="${OPTARG}"
+    ;;
+    d) domainUid="${OPTARG}"
+    ;;
+    r) restartVersion="${OPTARG}"
+    ;;
+    m) kubernetesCli="${OPTARG}"
+    ;;
+    v) verboseMode=true;
+    ;;
+    h) usage 0
+    ;;
+    *) usage 1
+    ;;
+  esac
+done
```

Comment:
Command line parsing problem: 

When I tried the above, it did not fail for "foo.sh somerandomstring" or "foo.sh -c myc somerandom -d foo". It only failed for "foo.sh -somerandomstring". 

This problem probably occurs in many of the scripts that use getopts. 

I assume "getopts" silently skips args that don't start with a "-".

Reply:
Yes. There are many scripts that use "getopts" (including create-domain.sh and others) and have similar behavior. I'm not sure if it's critical to fail when args that don't start with "-" is specified since those args will be ignored.

Reply:
Ok.